js/content/community.js

```diff
@@ -3670,42 +3671,75 @@ let WorkshopBrowseClass = (function(){
                 }`)
             }
 
             function changeSubscription(id) {
-                return new Promise(function(resolve) {
-                    let formData = new FormData();
-                    formData.append("sessionid", User.getSessionId());
-                    formData.append("appid", appid);
-                    formData.append("id", id);
-
-                    RequestData.post("https://steamcommunity.com/sharedfiles/" + method, formData, {
-                        withCredentials: true
-                    }).then(function() {
-                        updateWaitDialog();
-                        resolve();
-                    });
-                });
+                let formData = new FormData();
+                formData.append("sessionid", User.getSessionId());
+                formData.append("appid", appid);
+                formData.append("id", id);
+
+                return RequestData.post("https://steamcommunity.com/sharedfiles/" + method, formData, {
+                    withCredentials: true
+                }, true)
+                .then(function(res) {
+                    if (!res || !res.success || res.success !== 1) {
+                        failed++;
+                        console.error(new Error("Bad response"))
+                    }
+                    completed++;
+                    updateWaitDialog();
+                })
+                .catch(function(err) {
+                    failed++;
+                    completed++;
+                    updateWaitDialog();
+                    console.error(err);
+                })
             }
 
             Messenger.addMessageListener("startSubscriber", async function() {
                 updateWaitDialog();
 
+                function canSkip(method, node) {
+                    if (method === "subscribe") {
+                        return node && node.style.display !== "none";
+                    }
+
+                    if (method === "unsubscribe") {
+                        return !node || node.style.display === "none";
+                    }
+
+                    return false;
+                }
+
                 let workshopItems = [];
                 for (let p = 1; p <= Math.ceil(total / 30); p++) {
                     let url = new URL(window.location.href);
                     url.searchParams.set("p", p);
                     url.searchParams.set("numperpage", 30);
     
-                    let result = await RequestData.getHttp(url.toString());
-                    let xmlDoc = new DOMParser().parseFromString(result, "text/html");
+                    let result = await RequestData.getHttp(url.toString()).catch(err => console.error(err));
+                    if (!result) {
+                        await sleep(10000);
```

Comment:
I wouldn't retry in these cases, I'd log and continue

Reply:
Resolved in https://github.com/tfedor/AugmentedSteam/pull/506/commits/f58d20a58f438b82ee7a9e257498a535f161c4f2